standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5240,16 +5259,38 @@ public DropPartitionsResult drop_partitions_req(
         for (Path path : archToDelete) {
           wh.deleteDir(path, true, mustPurge, needsCm);
         }
+
+        // Uses a priority queue to delete the parents of deleted directories if empty.
+        // The parent with the largest size is always processed first. It guarantees that
```

Comment:
nit: `The parent with the largest size is always processed first ` -> `Parents with the deepest path are always processed first`

Reply:
Done.